lib/streamlit/runtime/app_session.py

```diff
@@ -479,7 +479,7 @@ def _on_scriptrunner_event(
         exception: BaseException | None = None,
         client_state: ClientState | None = None,
         page_script_hash: str | None = None,
-        fragment_ids_this_run: set[str] | None = None,
+        fragment_ids_this_run: list[str] | None = None,
```

Comment:
Ah, it's more because it's already in list form as the `fragment_id_queue` of a `RerunData` object, and I don't think there's much of a reason to convert it here since this argument is only copied into the proto message sent to the frontend.